### PR TITLE
Update to rust 1.53

### PR DESCRIPTION
**Pull request by Licenser**
# Pull request

## Description

Update to rust 1.53.0

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

## Performance

No measurable impact

### CODE REVIEW

File: src/codec/statsd.rs [hunk 1]
```diff
@@ -123,9 +123,7 @@ fn decode(data: &[u8], _ingest_ns: u64) -> Result<Value> {
         }
     }
     match d.next() {
-        Some((i, b'c')) | Some((i, b'h')) | Some((i, b's')) => {
-            m.insert("type".into(), substr(&data, i..=i)?.into())
-        }
+        Some((i, b'c' | b'h' | b's')) => m.insert("type".into(), substr(&data, i..=i)?.into()),
```
Comment (mfelsche) [inline src/codec/statsd.rs:126]: 😻

File: src/repository/artefact.rs [hunk 2]
```diff
@@ -465,6 +463,10 @@ impl Artefact for OnrampArtefact {
     }
 }
 
+impl Binding {
+    const LINKING_ERROR: &'static str = "links require the form of onramp -> pipeline or pipeline -> offramp or pipeline -> pipeline or pipeline -> onramp or offramp -> pipeline";
```
Comment (mfelsche) [inline src/repository/artefact.rs:467]: 💯

**Review by mfelsche — DISMISSED** [review]
YAY!

File: src/repository/artefact.rs (hunk 2 above)
Comment (Licenser) [reply src/repository/artefact.rs:467]: Ja I stole your habits :D they're now community property!

**Comment by mfelsche** [comment]
Lets hope that the coverage gods like the or patterns...

**Review by mfelsche — APPROVED** [review]

File: src/preprocessor.rs [hunk 3]
```diff
@@ -305,10 +305,7 @@ impl Preprocessor for Decompress {
                 decompressed
             }
             // ZLib magic headers
-            Some(&[0x78, 0x01, _, _, _, _])
-            | Some(&[0x78, 0x5e, _, _, _, _])
-            | Some(&[0x78, 0x9c, _, _, _, _])
-            | Some(&[0x78, 0xda, _, _, _, _]) => {
+            Some(&[0x78, 0x01 | 0x5e | 0x9c | 0xda, _, _, _, _]) => {
```
Comment (GaryPWhite) [inline src/preprocessor.rs:308]: incredible syntax change 😍

File: src/repository/artefact.rs [hunk 4]
```diff
@@ -494,7 +497,7 @@ impl Artefact for Binding {
             if let Some(inst) = src.instance() {
                 let mut instance = String::new();
                 for (map_name, map_replace) in &mappings {
-                    instance = inst.replace(&format!("%7B{}%7D", map_name), map_replace.as_str());
+                    instance = inst.replace(&format!("%7B{}%7D", map_name), &map_replace);
```
Comment (GaryPWhite) [inline src/repository/artefact.rs:500]: literally having struggled with `.as_str()` for exactly three PR's ever, this is still awesome 😄

Comment (Licenser) [reply src/repository/artefact.rs:500]: Ja those things keep changing and depending on what is done different syntaxes are needed there is `as_str`, `as_ref`, `deref`, `borrow`, `&` that all do slightly different but related things :D.

**Pull request merged**